File: Module2/3.Interfaces/src/Bird.java
public class Bird implements Flyable{
    private String name;
    private int age;

    public void fly(){
        System.out.println("Птица летит в небе. ");
    }

    public Bird(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public int getAge() {
        return age;
    }

    public String getName() {
        return name;
    }
}

File: Module4/4.10.OverridingMethodsProperties/src/Bird.kt
open class Bird(val type: String, val name: String, weight: Int, private val canFly: Boolean) {

    open fun fly(){
        when(canFly){
            true -> println("Птица летит")
            false -> println("Птица не может летать")
        }
    }

    open fun makeSound(){
        println("Птица издаёт звук")
    }
}

class Penguin(name: String): Bird("Penguin", name, 20, false){

    override fun fly() {
        println("Пингвины не летают!!!")
    }

    override fun makeSound() {
        println("Пингвин пищит")
    }
}

class Eagle(name: String): Bird("Eagle", name, 5, true){

    override fun fly(){
        println("Орёл летит высоко в небе.")
    }

    override fun makeSound() {
        println("Орёл издаёт крик")
    }
}